### PR TITLE
Remove specialist topic code from repo 1/4

### PR DESCRIPTION
**Pull request by georges1996**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

https://trello.com/c/RLc9saF6/2461-remove-specialist-topic-code-from-collections-publisher-m-l

This PR just removes most references to topics mainly cosmetic references in views and controllers and some usecases. This became unwieldy so moved the removal of the tag topic object and its related components here: https://github.com/alphagov/collections-publisher/pull/2102

### CODE REVIEW

**Review by beccapearce — APPROVED** [review]
Looks good!

And so many `if`s removed 💪🏻 woo

**Review by beccapearce — APPROVED** [review]
This is looking great! And seems to be working ok when testing locally :)
🚀

File: spec/services/tag_archiver_spec.rb [hunk 1]
```diff
@@ -1,168 +0,0 @@
-require "rails_helper"
-
-RSpec.describe TagArchiver do
```
Comment (hannako) [inline spec/services/tag_archiver_spec.rb:3]: Think we need to keep this spec file but remove the topic bits... some of it is still used ie the mainstream browse stuff

**Review by hannako — COMMENTED** [review]
Thanks George, this looks great! 

I couldn't figure out why you'd left loads of specialist topic code still in place until I realised you'd split it in to two PR's 🙈 ... So if it's not too much hassle could you update the PR description to say what this PR does, and what it does not and link to PR 2? 

Have you tested on integration to make sure it doesn't blow anything up? As the code looks good, but I've not tested it.

**Review by hannako — APPROVED** [review]
Thanks George!

**Review by anatron — APPROVED** [review]
LGTM!

**Pull request merged**